File: Models/ApsimCampbell/src/java/ApsimCampbell/SoiltempState.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;
public class SoiltempState
{
    private double netRadiation;
    private Double [] aveSoilWater;
    private Double [] bulkDensity;
    private double internalTimeStep;
    private Double [] thermalConductance;
    private Double [] thickness;
    private Boolean doInitialisationStuff;
    private double maxTempYesterday;
    private double timeOfDaySecs;
    private Integer numNodes;
    private Double [] soilWater;
    private Double [] clay;
    private Double [] soilTemp;
    private Double [] silt;
    private double instrumentHeight;
    private Double [] sand;
    private Integer numLayers;
    private Double [] volSpecHeatSoil;
    private double instrumHeight;
    private double canopyHeight;
    private Double [] heatStorage;
    private Double [] minSoilTemp;
    private Double [] maxSoilTemp;
    private Double [] newTemperature;
    private double airTemperature;
    private Double [] thermalConductivity;
    private double minTempYesterday;
    private Double [] carbon;
    private Double [] rocks;
    private Double [] InitialValues;
    private double boundaryLayerConductance;
    private Double [] aveSoilTemp;
    private Double [] morningSoilTemp;
    
    public SoiltempState() { }
    
    public SoiltempState(SoiltempState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.netRadiation = toCopy.getnetRadiation();
            aveSoilWater = new Double[toCopy.getaveSoilWater().length];
        for (int i = 0; i < toCopy.getaveSoilWater().length; i++)
        {
            aveSoilWater[i] = toCopy.getaveSoilWater()[i];
        }
            bulkDensity = new Double[toCopy.getbulkDensity().length];
        for (int i = 0; i < toCopy.getbulkDensity().length; i++)
        {
            bulkDensity[i] = toCopy.getbulkDensity()[i];
        }
            this.internalTimeStep = toCopy.getinternalTimeStep();
            thermalConductance = new Double[toCopy.getthermalConductance().length];
        for (int i = 0; i < toCopy.getthermalConductance().length; i++)
        {
            thermalConductance[i] = toCopy.getthermalConductance()[i];
        }
            thickness = new Double[toCopy.getthickness().length];
        for (int i = 0; i < toCopy.getthickness().length; i++)
        {
            thickness[i] = toCopy.getthickness()[i];
        }
            this.doInitialisationStuff = toCopy.getdoInitialisationStuff();
            this.maxTempYesterday = toCopy.getmaxTempYesterday();
            this.timeOfDaySecs = toCopy.gettimeOfDaySecs();
            this.numNodes = toCopy.getnumNodes();
            soilWater = new Double[toCopy.getsoilWater().length];
        for (int i = 0; i < toCopy.getsoilWater().length; i++)
        {
            soilWater[i] = toCopy.getsoilWater()[i];
        }
            clay = new Double[toCopy.getclay().length];
        for (int i = 0; i < toCopy.getclay().length; i++)
        {
            clay[i] = toCopy.getclay()[i];
        }
            soilTemp = new Double[toCopy.getsoilTemp().length];
        for (int i = 0; i < toCopy.getsoilTemp().length; i++)
        {
            soilTemp[i] = toCopy.getsoilTemp()[i];
        }
            silt = new Double[toCopy.getsilt().length];
        for (int i = 0; i < toCopy.getsilt().length; i++)
        {
            silt[i] = toCopy.getsilt()[i];
        }
            this.instrumentHeight = toCopy.getinstrumentHeight();
            sand = new Double[toCopy.getsand().length];
        for (int i = 0; i < toCopy.getsand().length; i++)
        {
            sand[i] = toCopy.getsand()[i];
        }
            this.numLayers = toCopy.getnumLayers();
            volSpecHeatSoil = new Double[toCopy.getvolSpecHeatSoil().length];
        for (int i = 0; i < toCopy.getvolSpecHeatSoil().length; i++)
        {
            volSpecHeatSoil[i] = toCopy.getvolSpecHeatSoil()[i];
        }
            this.instrumHeight = toCopy.getinstrumHeight();
            this.canopyHeight = toCopy.getcanopyHeight();
            heatStorage = new Double[toCopy.getheatStorage().length];
        for (int i = 0; i < toCopy.getheatStorage().length; i++)
        {
            heatStorage[i] = toCopy.getheatStorage()[i];
        }
            minSoilTemp = new Double[toCopy.getminSoilTemp().length];
        for (int i = 0; i < toCopy.getminSoilTemp().length; i++)
        {
            minSoilTemp[i] = toCopy.getminSoilTemp()[i];
        }
            maxSoilTemp = new Double[toCopy.getmaxSoilTemp().length];
        for (int i = 0; i < toCopy.getmaxSoilTemp().length; i++)
        {
            maxSoilTemp[i] = toCopy.getmaxSoilTemp()[i];
        }
            newTemperature = new Double[toCopy.getnewTemperature().length];
        for (int i = 0; i < toCopy.getnewTemperature().length; i++)
        {
            newTemperature[i] = toCopy.getnewTemperature()[i];
        }
            this.airTemperature = toCopy.getairTemperature();
            thermalConductivity = new Double[toCopy.getthermalConductivity().length];
        for (int i = 0; i < toCopy.getthermalConductivity().length; i++)
        {
            thermalConductivity[i] = toCopy.getthermalConductivity()[i];
        }
            this.minTempYesterday = toCopy.getminTempYesterday();
            carbon = new Double[toCopy.getcarbon().length];
        for (int i = 0; i < toCopy.getcarbon().length; i++)
        {
            carbon[i] = toCopy.getcarbon()[i];
        }
            rocks = new Double[toCopy.getrocks().length];
        for (int i = 0; i < toCopy.getrocks().length; i++)
        {
            rocks[i] = toCopy.getrocks()[i];
        }
            InitialValues = new Double[toCopy.getInitialValues().length];
        for (int i = 0; i < toCopy.getInitialValues().length; i++)
        {
            InitialValues[i] = toCopy.getInitialValues()[i];
        }
            this.boundaryLayerConductance = toCopy.getboundaryLayerConductance();
            aveSoilTemp = new Double[toCopy.getaveSoilTemp().length];
        for (int i = 0; i < toCopy.getaveSoilTemp().length; i++)
        {
            aveSoilTemp[i] = toCopy.getaveSoilTemp()[i];
        }
            morningSoilTemp = new Double[toCopy.getmorningSoilTemp().length];
        for (int i = 0; i < toCopy.getmorningSoilTemp().length; i++)
        {
            morningSoilTemp[i] = toCopy.getmorningSoilTemp()[i];
        }
        }
    }
    public double getnetRadiation()
    { return netRadiation; }

    public void setnetRadiation(double _netRadiation)
    { this.netRadiation= _netRadiation; } 
    
    public Double [] getaveSoilWater()
    { return aveSoilWater; }

    public void setaveSoilWater(Double [] _aveSoilWater)
    { this.aveSoilWater= _aveSoilWater; } 
    
    public Double [] getbulkDensity()
    { return bulkDensity; }

    public void setbulkDensity(Double [] _bulkDensity)
    { this.bulkDensity= _bulkDensity; } 
    
    public double getinternalTimeStep()
    { return internalTimeStep; }

    public void setinternalTimeStep(double _internalTimeStep)
    { this.internalTimeStep= _internalTimeStep; } 
    
    public Double [] getthermalConductance()
    { return thermalConductance; }

    public void setthermalConductance(Double [] _thermalConductance)
    { this.thermalConductance= _thermalConductance; } 
    
    public Double [] getthickness()
    { return thickness; }

    public void setthickness(Double [] _thickness)
    { this.thickness= _thickness; } 
    
    public Boolean getdoInitialisationStuff()
    { return doInitialisationStuff; }

    public void setdoInitialisationStuff(Boolean _doInitialisationStuff)
    { this.doInitialisationStuff= _doInitialisationStuff; } 
    
    public double getmaxTempYesterday()
    { return maxTempYesterday; }

    public void setmaxTempYesterday(double _maxTempYesterday)
    { this.maxTempYesterday= _maxTempYesterday; } 
    
    public double gettimeOfDaySecs()
    { return timeOfDaySecs; }

    public void settimeOfDaySecs(double _timeOfDaySecs)
    { this.timeOfDaySecs= _timeOfDaySecs; } 
    
    public Integer getnumNodes()
    { return numNodes; }

    public void setnumNodes(Integer _numNodes)
    { this.numNodes= _numNodes; } 
    
    public Double [] getsoilWater()
    { return soilWater; }

    public void setsoilWater(Double [] _soilWater)
    { this.soilWater= _soilWater; } 
    
    public Double [] getclay()
    { return clay; }

    public void setclay(Double [] _clay)
    { this.clay= _clay; } 
    
    public Double [] getsoilTemp()
    { return soilTemp; }

    public void setsoilTemp(Double [] _soilTemp)
    { this.soilTemp= _soilTemp; } 
    
    public Double [] getsilt()
    { return silt; }

    public void setsilt(Double [] _silt)
    { this.silt= _silt; } 
    
    public double getinstrumentHeight()
    { return instrumentHeight; }

    public void setinstrumentHeight(double _instrumentHeight)
    { this.instrumentHeight= _instrumentHeight; } 
    
    public Double [] getsand()
    { return sand; }

    public void setsand(Double [] _sand)
    { this.sand= _sand; } 
    
    public Integer getnumLayers()
    { return numLayers; }

    public void setnumLayers(Integer _numLayers)
    { this.numLayers= _numLayers; } 
    
    public Double [] getvolSpecHeatSoil()
    { return volSpecHeatSoil; }

    public void setvolSpecHeatSoil(Double [] _volSpecHeatSoil)
    { this.volSpecHeatSoil= _volSpecHeatSoil; } 
    
    public double getinstrumHeight()
    { return instrumHeight; }

    public void setinstrumHeight(double _instrumHeight)
    { this.instrumHeight= _instrumHeight; } 
    
    public double getcanopyHeight()
    { return canopyHeight; }

    public void setcanopyHeight(double _canopyHeight)
    { this.canopyHeight= _canopyHeight; } 
    
    public Double [] getheatStorage()
    { return heatStorage; }

    public void setheatStorage(Double [] _heatStorage)
    { this.heatStorage= _heatStorage; } 
    
    public Double [] getminSoilTemp()
    { return minSoilTemp; }

    public void setminSoilTemp(Double [] _minSoilTemp)
    { this.minSoilTemp= _minSoilTemp; } 
    
    public Double [] getmaxSoilTemp()
    { return maxSoilTemp; }

    public void setmaxSoilTemp(Double [] _maxSoilTemp)
    { this.maxSoilTemp= _maxSoilTemp; } 
    
    public Double [] getnewTemperature()
    { return newTemperature; }

    public void setnewTemperature(Double [] _newTemperature)
    { this.newTemperature= _newTemperature; } 
    
    public double getairTemperature()
    { return airTemperature; }

    public void setairTemperature(double _airTemperature)
    { this.airTemperature= _airTemperature; } 
    
    public Double [] getthermalConductivity()
    { return thermalConductivity; }

    public void setthermalConductivity(Double [] _thermalConductivity)
    { this.thermalConductivity= _thermalConductivity; } 
    
    public double getminTempYesterday()
    { return minTempYesterday; }

    public void setminTempYesterday(double _minTempYesterday)
    { this.minTempYesterday= _minTempYesterday; } 
    
    public Double [] getcarbon()
    { return carbon; }

    public void setcarbon(Double [] _carbon)
    { this.carbon= _carbon; } 
    
    public Double [] getrocks()
    { return rocks; }

    public void setrocks(Double [] _rocks)
    { this.rocks= _rocks; } 
    
    public Double [] getInitialValues()
    { return InitialValues; }

    public void setInitialValues(Double [] _InitialValues)
    { this.InitialValues= _InitialValues; } 
    
    public double getboundaryLayerConductance()
    { return boundaryLayerConductance; }

    public void setboundaryLayerConductance(double _boundaryLayerConductance)
    { this.boundaryLayerConductance= _boundaryLayerConductance; } 
    
    public Double [] getaveSoilTemp()
    { return aveSoilTemp; }

    public void setaveSoilTemp(Double [] _aveSoilTemp)
    { this.aveSoilTemp= _aveSoilTemp; } 
    
    public Double [] getmorningSoilTemp()
    { return morningSoilTemp; }

    public void setmorningSoilTemp(Double [] _morningSoilTemp)
    { this.morningSoilTemp= _morningSoilTemp; } 
    
}

File: Models/ApsimCampbell/src/cs/ApsimCampbell/SoiltempState.cs
using System;
using System.Collections.Generic;
public class SoiltempState 
{
    private double _minTempYesterday;
    private double[] _minSoilTemp;
    private double[] _newTemperature;
    private bool _doInitialisationStuff;
    private double[] _silt;
    private int _numNodes;
    private double[] _morningSoilTemp;
    private double[] _aveSoilTemp;
    private double[] _clay;
    private double[] _soilWater;
    private double[] _thermalConductivity;
    private double _instrumHeight;
    private double[] _aveSoilWater;
    private double[] _carbon;
    private double[] _soilTemp;
    private double[] _thickness;
    private double[] _heatStorage;
    private double _instrumentHeight;
    private double[] _thermalConductance;
    private double _maxTempYesterday;
    private double _boundaryLayerConductance;
    private double _canopyHeight;
    private double _netRadiation;
    private double[] _maxSoilTemp;
    private double[] _volSpecHeatSoil;
    private double[] _rocks;
    private double[] _bulkDensity;
    private int _numLayers;
    private double _airTemperature;
    private double[] _sand;
    private double _timeOfDaySecs;
    private double _internalTimeStep;
    
    /// <summary>
    /// Constructor of the SoiltempState component")
    /// </summary>  
    public SoiltempState() { }
    
    
    public SoiltempState(SoiltempState toCopy, bool copyAll) // copy constructor 
    {
        if (copyAll)
        {
    
            minTempYesterday = toCopy.minTempYesterday;
            minSoilTemp = new double[toCopy.minSoilTemp.Length];
        for (int i = 0; i < toCopy.minSoilTemp.Length; i++)
            { minSoilTemp[i] = toCopy.minSoilTemp[i]; }
    
            newTemperature = new double[toCopy.newTemperature.Length];
        for (int i = 0; i < toCopy.newTemperature.Length; i++)
            { newTemperature[i] = toCopy.newTemperature[i]; }
    
            doInitialisationStuff = toCopy.doInitialisationStuff;
            silt = new double[toCopy.silt.Length];
        for (int i = 0; i < toCopy.silt.Length; i++)
            { silt[i] = toCopy.silt[i]; }
    
            numNodes = toCopy.numNodes;
            morningSoilTemp = new double[toCopy.morningSoilTemp.Length];
        for (int i = 0; i < toCopy.morningSoilTemp.Length; i++)
            { morningSoilTemp[i] = toCopy.morningSoilTemp[i]; }
    
            aveSoilTemp = new double[toCopy.aveSoilTemp.Length];
        for (int i = 0; i < toCopy.aveSoilTemp.Length; i++)
            { aveSoilTemp[i] = toCopy.aveSoilTemp[i]; }
    
            clay = new double[toCopy.clay.Length];
        for (int i = 0; i < toCopy.clay.Length; i++)
            { clay[i] = toCopy.clay[i]; }
    
            soilWater = new double[toCopy.soilWater.Length];
        for (int i = 0; i < toCopy.soilWater.Length; i++)
            { soilWater[i] = toCopy.soilWater[i]; }
    
            thermalConductivity = new double[toCopy.thermalConductivity.Length];
        for (int i = 0; i < toCopy.thermalConductivity.Length; i++)
            { thermalConductivity[i] = toCopy.thermalConductivity[i]; }
    
            instrumHeight = toCopy.instrumHeight;
            aveSoilWater = new double[toCopy.aveSoilWater.Length];
        for (int i = 0; i < toCopy.aveSoilWater.Length; i++)
            { aveSoilWater[i] = toCopy.aveSoilWater[i]; }
    
            carbon = new double[toCopy.carbon.Length];
        for (int i = 0; i < toCopy.carbon.Length; i++)
            { carbon[i] = toCopy.carbon[i]; }
    
            soilTemp = new double[toCopy.soilTemp.Length];
        for (int i = 0; i < toCopy.soilTemp.Length; i++)
            { soilTemp[i] = toCopy.soilTemp[i]; }
    
            thickness = new double[toCopy.thickness.Length];
        for (int i = 0; i < toCopy.thickness.Length; i++)
            { thickness[i] = toCopy.thickness[i]; }
    
            heatStorage = new double[toCopy.heatStorage.Length];
        for (int i = 0; i < toCopy.heatStorage.Length; i++)
            { heatStorage[i] = toCopy.heatStorage[i]; }
    
            instrumentHeight = toCopy.instrumentHeight;
            thermalConductance = new double[toCopy.thermalConductance.Length];
        for (int i = 0; i < toCopy.thermalConductance.Length; i++)
            { thermalConductance[i] = toCopy.thermalConductance[i]; }
    
            maxTempYesterday = toCopy.maxTempYesterday;
            boundaryLayerConductance = toCopy.boundaryLayerConductance;
            canopyHeight = toCopy.canopyHeight;
            netRadiation = toCopy.netRadiation;
            maxSoilTemp = new double[toCopy.maxSoilTemp.Length];
        for (int i = 0; i < toCopy.maxSoilTemp.Length; i++)
            { maxSoilTemp[i] = toCopy.maxSoilTemp[i]; }
    
            volSpecHeatSoil = new double[toCopy.volSpecHeatSoil.Length];
        for (int i = 0; i < toCopy.volSpecHeatSoil.Length; i++)
            { volSpecHeatSoil[i] = toCopy.volSpecHeatSoil[i]; }
    
            rocks = new double[toCopy.rocks.Length];
        for (int i = 0; i < toCopy.rocks.Length; i++)
            { rocks[i] = toCopy.rocks[i]; }
    
            bulkDensity = new double[toCopy.bulkDensity.Length];
        for (int i = 0; i < toCopy.bulkDensity.Length; i++)
            { bulkDensity[i] = toCopy.bulkDensity[i]; }
    
            numLayers = toCopy.numLayers;
            airTemperature = toCopy.airTemperature;
            sand = new double[toCopy.sand.Length];
        for (int i = 0; i < toCopy.sand.Length; i++)
            { sand[i] = toCopy.sand[i]; }
    
            timeOfDaySecs = toCopy.timeOfDaySecs;
            internalTimeStep = toCopy.internalTimeStep;
        }
    }
    public double minTempYesterday
    {
        get { return this._minTempYesterday; }
        set { this._minTempYesterday= value; } 
    }
    public double[] minSoilTemp
    {
        get { return this._minSoilTemp; }
        set { this._minSoilTemp= value; } 
    }
    public double[] newTemperature
    {
        get { return this._newTemperature; }
        set { this._newTemperature= value; } 
    }
    public bool doInitialisationStuff
    {
        get { return this._doInitialisationStuff; }
        set { this._doInitialisationStuff= value; } 
    }
    public double[] silt
    {
        get { return this._silt; }
        set { this._silt= value; } 
    }
    public int numNodes
    {
        get { return this._numNodes; }
        set { this._numNodes= value; } 
    }
    public double[] morningSoilTemp
    {
        get { return this._morningSoilTemp; }
        set { this._morningSoilTemp= value; } 
    }
    public double[] aveSoilTemp
    {
        get { return this._aveSoilTemp; }
        set { this._aveSoilTemp= value; } 
    }
    public double[] clay
    {
        get { return this._clay; }
        set { this._clay= value; } 
    }
    public double[] soilWater
    {
        get { return this._soilWater; }
        set { this._soilWater= value; } 
    }
    public double[] thermalConductivity
    {
        get { return this._thermalConductivity; }
        set { this._thermalConductivity= value; } 
    }
    public double instrumHeight
    {
        get { return this._instrumHeight; }
        set { this._instrumHeight= value; } 
    }
    public double[] aveSoilWater
    {
        get { return this._aveSoilWater; }
        set { this._aveSoilWater= value; } 
    }
    public double[] carbon
    {
        get { return this._carbon; }
        set { this._carbon= value; } 
    }
    public double[] soilTemp
    {
        get { return this._soilTemp; }
        set { this._soilTemp= value; } 
    }
    public double[] thickness
    {
        get { return this._thickness; }
        set { this._thickness= value; } 
    }
    public double[] heatStorage
    {
        get { return this._heatStorage; }
        set { this._heatStorage= value; } 
    }
    public double instrumentHeight
    {
        get { return this._instrumentHeight; }
        set { this._instrumentHeight= value; } 
    }
    public double[] thermalConductance
    {
        get { return this._thermalConductance; }
        set { this._thermalConductance= value; } 
    }
    public double maxTempYesterday
    {
        get { return this._maxTempYesterday; }
        set { this._maxTempYesterday= value; } 
    }
    public double boundaryLayerConductance
    {
        get { return this._boundaryLayerConductance; }
        set { this._boundaryLayerConductance= value; } 
    }
    public double canopyHeight
    {
        get { return this._canopyHeight; }
        set { this._canopyHeight= value; } 
    }
    public double netRadiation
    {
        get { return this._netRadiation; }
        set { this._netRadiation= value; } 
    }
    public double[] maxSoilTemp
    {
        get { return this._maxSoilTemp; }
        set { this._maxSoilTemp= value; } 
    }
    public double[] volSpecHeatSoil
    {
        get { return this._volSpecHeatSoil; }
        set { this._volSpecHeatSoil= value; } 
    }
    public double[] rocks
    {
        get { return this._rocks; }
        set { this._rocks= value; } 
    }
    public double[] bulkDensity
    {
        get { return this._bulkDensity; }
        set { this._bulkDensity= value; } 
    }
    public int numLayers
    {
        get { return this._numLayers; }
        set { this._numLayers= value; } 
    }
    public double airTemperature
    {
        get { return this._airTemperature; }
        set { this._airTemperature= value; } 
    }
    public double[] sand
    {
        get { return this._sand; }
        set { this._sand= value; } 
    }
    public double timeOfDaySecs
    {
        get { return this._timeOfDaySecs; }
        set { this._timeOfDaySecs= value; } 
    }
    public double internalTimeStep
    {
        get { return this._internalTimeStep; }
        set { this._internalTimeStep= value; } 
    }
}